src/twisted/internet/defer.py

```diff
@@ -2003,7 +2003,9 @@ def __aenter__(self: _ConcurrencyPrimitiveT) -> Deferred[_ConcurrencyPrimitiveT]
         """
         return self.acquire()
 
-    def __aexit__(self, exc_type: bool, exc_val: bool, exc_tb: bool) -> Deferred[bool]:
+    def __aexit__(
+        self, exc_type: bool, exc_val: bool, exc_tb: bool
+    ) -> Deferred[Literal[False]]:
```

Comment:
I note in passing that the annotations for `exc_type`, `exc_val` and `exc_tb` types look wrong here, see e.g. https://github.com/python/typeshed/blob/a7cf6144deeb3f99570f8b97fb5f5ecf16f20c01/stdlib/contextlib.pyi#L52-L54

Reply:
A good point in favor of having some kind of test for these changes.

Reply:
These annotations still look wrong.

Reply:
Oh sorry. I was focusing so much on the testing I missed this comment.  I can update that, but I'm not sure how to make mypy complain about this annotation in order to "test" it.

Reply:
Nor I :/

Reply:
Done!